winsup/cygwin/dcrt0.cc

```diff
@@ -545,7 +545,7 @@ get_cygwin_startup_info ()
   child_info *res = (child_info *) si.lpReserved2;
 
   if (si.cbReserved2 < EXEC_MAGIC_SIZE || !res
-      || res->intro != PROC_MAGIC_GENERIC || res->magic != CHILD_INFO_MAGIC)
+      || res->intro != PROC_MAGIC_GENERIC || res->magic != (CHILD_INFO_MAGIC ^ CYGWIN_VERSION_DLL_COMBINED))
```

Comment:
I was surprised at first at the use of bitwise XOR here, as I would expect this to be a bitwise OR if these were flags. However, [`CYGWIN_VERSION_DLL_COMBINED` is an integer with multiple bits](https://github.com/git-for-windows/msys2-runtime/blob/54f02c59d56a03cab955c04b3d2dfe6b1ccfc171/winsup/cygwin/include/cygwin/version.h#L20-L27) (currently set to `3003006` to represent `3.3.6`).

So this "magic" is very magical, it seems (and you do a similar update in `sigproc.cc`.

Reply:
But the commit message motivates this well, saying that we should use a new magic value whenever Git for Windows' version of MSYS2 is updated. (It's not accurate that only a new Git version would change this magic, but if two Git for Windows releases have the identical MSYS2 version, then they should be compatible still.)

Reply:
Indeed. The idea here is to have some kind of unique identifier that won't ever be shared between different MSYS2 runtime versions. Hence the XOR (because the OR would decrease the entropy).